workflow/context.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2024 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package workflow
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/microsoft/durabletask-go/task"
+)
+
+type WorkflowContext struct {
+	orchestrationContext *task.OrchestrationContext
+}
+
+// GetInput casts the input from the context to a specified interface.
+func (wfc *WorkflowContext) GetInput(v interface{}) error {
+	return wfc.orchestrationContext.GetInput(&v)
+}
+
+// Name returns the name string from the workflow context.
+func (wfc *WorkflowContext) Name() string {
+	return wfc.orchestrationContext.Name
+}
+
+// InstanceID returns the ID of the currently executing workflow
+func (wfc *WorkflowContext) InstanceID() string {
+	return fmt.Sprintf("%v", wfc.orchestrationContext.ID)
+}
+
+// CurrentUTCDateTime returns the current workflow time as UTC. Note that this should be used instead of `time.Now()`, which is not compatible with workflow replays.
+func (wfc *WorkflowContext) CurrentUTCDateTime() time.Time {
+	return wfc.orchestrationContext.CurrentTimeUtc
+}
+
+// IsReplaying returns whether the workflow is replaying.
+func (wfc *WorkflowContext) IsReplaying() bool {
+	return wfc.orchestrationContext.IsReplaying
+}
+
+// CallActivity returns a completable task for a given activity.
+// NOTE: The task must be invoked with the .Await(output interface{}) method.
```

Comment:
Regarding these NOTEs, users can choose to not await the tasks at call time. This is most common when doing the fan-out/fan-in pattern, where you want multiple tasks to run in parallel. I wonder if we should change these comments.

Reply:
I think the note update improves clarity slightly